Fix vagrant so that ssh commands work OOTB

jayunit100 opened the pull request:
This adds a provisioner specific unit testing framework, so that any provisioner can test its specific functionality. 

It also fixes a particular bug for a particular provisioner which im particularly fond of :)

jayunit100 committed:
Fix vagrant so that ssh commands work OOTB, (squashed) move verify to vagrant/util.sh, remove run_provider_test, cleanup.

derekwaynecarr commented on cluster/validate-cluster.sh:
```diff
@@ -102,4 +102,10 @@ for (( i=0; i<${#MINION_NAMES[@]}; i++)); do
+# Here an individual provider can add extra tests, 
+# For example 'vagrant ssh master1 kubectl get pods' should work
+# to show existing pods, and so on.
+${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}/validate-provider.sh
```
You would need a shell file for each provider?

My bias is we do not do this piece since the goal is to move more of this into a Go program.

derekwaynecarr commented on cluster/vagrant/provision-master.sh:
```diff
@@ -54,6 +54,7 @@ for (( i=0; i<${#MINION_NAMES[@]}; i++)); do
     echo "Adding $minion to hosts file"
     echo "$ip $minion" >> /etc/hosts
   fi
+  echo "127.0.0.1 localhost" >> /etc/hosts
```
You will want this in provision-minion as well?

jayunit100 replied:
good point derek.  I don't think its needed in minions.

jayunit100 replied:
oh, never mind, I thought you were  saying "why did you put this in provision-minion.sh"... 

the answer is *no* I don't need this in minion, the expectation of the test is that the master is able to run "get pods" against itself.  Wether or not the minions are running this command is not needed....

makes sense?

jayunit100 commented:
looks like build has passed now, i can rebase if you think its a good idea.

derekwaynecarr commented:
No need to rebase.  I am not sure I understand the value of the validate-cluster operation per provider right now as presented.  It looks like something that will just make things slower.  Is there any reason you can't move whatever additional verification you want to do in cluster/vagrant/util.sh verify-cluster function?

jayunit100 commented:
you're right - that is a great place for it  *cluster/vagrant/util.sh* , which seems to be serving as an interface .   Somehow that completely skipped my mind .

derekwaynecarr commented:
squash your commits

jayunit100 commented:
squashed a few and minimized the change, removed one extra file.  I think this is something we can agree on.  if this looks useful feel free to push, if not happy to close also....

derekwaynecarr commented:
I am not seeing the squash... 

I still see 5 commits where 1 commit would be appropriate - you are just changing 2 files and 7 lines ;-)

jayunit100 commented:
hmmm funny, let me push again

jayunit100 commented:
okay i just rebased /pushed  :)

derekwaynecarr commented:
Thanks ;-)
